src/Core/TypeConverter.fs

```diff
@@ -0,0 +1,25 @@
+module TypeConverter
```

Comment:
So... I think I get a rough idea how conjugation serialization is working here. But I am not sure if this is the way to go. Things like tuple converters and type descriptors bring new concepts to the code and complicate it, they make serialization and deserialization even less uniform and they are hard to autotest. 

As a result we have all the forms in one column though I don't see much advantage about that... Table becomes less readable and less filterable. Another thing is that we make our life more complicated if we want to add some regularity filters (e.g. form `jedí` for `jíst` is exceptional). We also thereby drift even further away from the idea of having one table for all the verb exercises - they all now basically utilize one wiki table taking different forms from there.

I actually really appreciate this attempt because I wanted to try this approach as well to see what it takes. Still I would suggest just plainly adding six columns with different forms to the table and fill them in the way we fill everything else.

Reply:
is there a way to store data in document format? (mongodb style?). It seems reasonable to split columns from an SQL perspective, but I would keep those in one if it's possible to query data based on subdocument properties.

Reply:
Err... I have not worked with MongoDB, yet generally [this](https://docs.microsoft.com/en-us/rest/api/storageservices/understanding-the-table-service-data-model#property-types) is all we have, Azure Table Storage is very basic.

Azure has a thing called [Cosmos DB](https://docs.microsoft.com/en-us/azure/cosmos-db/introduction) which is a more profound thing with multiple APIs and definitely does something with [MongoDB](https://docs.microsoft.com/en-us/azure/cosmos-db/faq#azure-cosmos-dbs-api-for-mongodb).

I technically don't mind Cosmos DB in the future, but moving there takes effort and money so it should be a very weighed and organized decision...